sdk/search/Azure.AI.Search/tests/Utilities/TestLogger.cs

```diff
@@ -0,0 +1,160 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.Tracing;
+using System.Linq;
+using System.Text;
+using Azure.Core.Diagnostics;
+using NUnit.Framework;
+using NUnit.Framework.Interfaces;
+
+namespace Azure.AI.Search.Tests
+{
+    /// <summary>
+    /// The TestLogger listens for the AzureSDK logging event source and traces
+    /// the output so it's easy to view the logs when testing.
+    ///
+    /// We create an instance in SearchTestBase when we want to log test
+    /// output.
+    /// </summary>
+    internal class TestLogger : IDisposable
```

Comment:
Is this a copy of storage logger? Should we move it to shared?

Reply:
Yes - I renamed it from `TestEventListener` so it'll be easier to move into the shared testing fx in the near future.